docs/specification/2025-03-26/basic/authorization.mdx

```diff
@@ -63,248 +156,329 @@ audience. For instance:
-**MUST** use the following default endpoint paths relative to the authorization base URL
-(as defined in [Section 2.3.2](#232-authorization-base-url)):
+MCP servers that implement their own AS **MUST** host appropriate AS metadata, as outlined in
+[RFC 8414 Section 3](https://datatracker.ietf.org/doc/html/rfc8414#section-3).
 
-| Endpoint               | Default Path | Description                          |
-| ---------------------- | ------------ | ------------------------------------ |
-| Authorization Endpoint | /authorize   | Used for authorization requests      |
-| Token Endpoint         | /token       | Used for token exchange & refresh    |
-| Registration Endpoint  | /register    | Used for dynamic client registration |
+MCP servers **MAY** support delegated authorization through third-party AS.
+In this flow, the MCP server acts as both an OAuth client (to the external AS) and an OAuth
+AS (to the MCP client).
 
-For example, with an MCP server hosted at `https://api.example.com/v1/mcp`, the default
-endpoints would be:
+#### 2.5.1 Special Considerations
 
-- `https://api.example.com/authorize`
-- `https://api.example.com/token`
-- `https://api.example.com/register`
+MCP servers that use delegated authorization or implement a colocated AS **MUST**:
 
-Clients **MUST** first attempt to discover endpoints via the metadata document before
-falling back to default paths. When using default paths, all other protocol requirements
-remain unchanged.
+1. Maintain secure mapping between third-party tokens and issued client sessions.
+1. Never pass third-party tokens directly to the client.
+1. Validate token status before granting any data access.
+1. Implement appropriate token lifecycle management.  
+1. Provide appropriate responses to the client if re-authentication is needed. Please refer to
+   [RFC 9470 OAuth 2.0 Step Up Authentication Challenge Protocol](https://www.rfc-editor.org/rfc/rfc9470.html).
 
-### 2.4 Dynamic Client Registration
+### 2.7 Dynamic Client Registration
 
-MCP clients and servers **SHOULD** support the
-[OAuth 2.0 Dynamic Client Registration Protocol](https://datatracker.ietf.org/doc/html/rfc7591)
-to allow MCP clients to obtain OAuth client IDs without user interaction. This provides a
-standardized way for clients to automatically register with new servers, which is crucial
-for MCP because:
+MCP clients **SHOULD** support the
+[OAuth 2.0 Dynamic Client Registration Protocol](https://datatracker.ietf.org/doc/html/rfc7591) to
+allow MCP clients to obtain OAuth client IDs without user interaction. This provides a standardized
+way for clients to automatically register with new servers, which is crucial for MCP because:
 
-- Clients cannot know all possible servers in advance
-- Manual registration would create friction for users
-- It enables seamless connection to new servers
+- Clients cannot know all possible servers in advance  
+- Manual registration would create friction for users  
```

Comment:
Should say "friction for developers"? Users are only seeing the final product anyway.

Reply:
It's both, because in some alternative view there is the possibility that client users _can_ provide the client ID they want to use with the server (as rough as that approach is from a usability perspective). I updated the verbiage for it.